go/engine/signup.go

```diff
@@ -234,12 +234,14 @@ func (s *SignupEngine) registerDevice(a libkb.LoginContext, ctx *Context, device
 		s.G().Log.Warning("error saving session file: %s", err)
 	}
 
+	// Create the secret store as late as possible here, as the username may
+	// change during the signup process.
 	if s.arg.StoreSecret {
-		// Create the secret store as late as possible here
-		// (instead of when we first get the value of
-		// StoreSecret) as the username may change during the
-		// signup process.
 		secretStore := libkb.NewSecretStore(s.G(), s.me.GetNormalizedName())
+		if secretStore == nil {
+			s.G().Log.Info("aborting StoreSecret: secretStore is nil")
+			return nil
+		}
```

Comment:
Came in this commit: https://github.com/keybase/client/pull/7083/commits/c3e8a1d7315c33fe44bb5a4802f74a3ee09365d5 I'm guessing it was to have a clear failure instead of crashing on nil-pointer?